authentication/models.py

```diff
@@ -0,0 +1,18 @@
+from django.db import models
+from django.contrib.auth.models import AbstractUser
+# Create your models here.
+
+class User(AbstractUser):
+  uid = models.CharField(primary_key=True, max_length=64)
+  username = models.CharField(unique=True, blank=True, max_length=64)
```

Comment:
This line will give an error when there are two or more users without a username, since the uniqueness constraint on the username will be violated (Blank values are considered to be the same). As the null values aren't taken into account in uniqueness constraints, we can keep the username to be null by default.

```suggestion
  username = models.CharField(default=None, unique=True, null=True, max_length=64)
```

Another method would be to remove the uniqueness constraint here and check this in the serializers when a user updates the username. Either of the two methods would work.